thread_pthread.c

```diff
@@ -2451,9 +2451,12 @@ rb_thread_execute_hooks(rb_event_flag_t event)
 
     if (rb_internal_thread_event_hooks) {
         rb_internal_thread_event_hook_t *h = rb_internal_thread_event_hooks;
+        rb_internal_thread_event_data_t *data = ALLOC_N(rb_internal_thread_event_data_t, 1);
+        data->thread = th->self;
```

Comment:
Why dynamically allocating?
This seems a memory leak.

```suggestion
        rb_internal_thread_event_data_t data[1] = {{.thread = th->self}};
```